mlflow/genai/judges/utils.py

```diff
@@ -77,6 +127,108 @@ def _sanitize_justification(justification: str) -> str:
     return justification.replace("Let's think step by step. ", "")
 
 
+def _invoke_databricks_judge(
+    prompt: str | list["ChatMessage"],
+    assessment_name: str,
+) -> Feedback:
+    """
+    Invoke the Databricks judge using the databricks.agents.evals library.
+
+    Args:
+        prompt: The formatted prompt with template variables filled in.
+        assessment_name: The name of the assessment.
+
+    Returns:
+        Feedback object from the Databricks judge.
+
+    Raises:
+        MlflowException: If databricks-agents is not installed or invocation fails.
+    """
+    try:
+        from databricks.agents.evals.judges import guidelines
+
+        # Extract the instructions/guidelines and context from the prompt
+        # The prompt has already been formatted with template variables filled in
+        instructions_text = ""
+        context = {}
+
+        if isinstance(prompt, str):
+            instructions_text = prompt
+            context["evaluation_prompt"] = prompt
+        else:
+            if isinstance(prompt, list) and len(prompt) > 0:
+                for msg in prompt:
+                    if isinstance(msg, ChatMessage):
+                        if msg.role == "system":
+                            instructions_text = msg.content
+                        elif msg.role == "user":
+                            context["evaluation_data"] = msg.content
+
+                if not instructions_text:
+                    messages_content = []
+                    for msg in prompt:
+                        if isinstance(msg, ChatMessage):
+                            messages_content.append(f"{msg.role}: {msg.content}")
+                    instructions_text = "\n".join(messages_content)
+
+        return guidelines(
+            guidelines=instructions_text,
+            context=context,
+            assessment_name=assessment_name,
+        )
```

Comment:
The guidelines judge comes with its own built-in prompt that doesn't necessarily match the instructions the user provided. @alkispoly-db @smoorjani is there a way to call the judge API directly? seems like the approach that we use for custom prompt judge is likely what we want to use here.

Reply:
Custom prompt sounds about right. I'll update to use that instead